Rules/CommonScripts/plants_regrow.as

```diff
@@ -0,0 +1,199 @@
+
+	TileInfo() {};
+
+	TileInfo(Vec2f _coords, u32 _place_time, Tile _tile)
+	{
+		coords = _coords;
+		place_time = _place_time;
+		tile = _tile;
+	};
+
+	bool mossTime()
+	{
+		return (getGameTime() - place_time > moss_time_min);
+	}
+
+	u8 grassLuck()
+	{
+		u8 luck = 0;
+		CMap@ map = getMap();
+
+		Tile tile_left = map.getTile(coords-Vec2f(map.tilesize, map.tilesize));
+		Tile tile_right = map.getTile(coords-Vec2f(-map.tilesize, map.tilesize));
+
+		if (tile_left.type == CMap::tile_grass) luck += 15;
+		if (tile_right.type == CMap::tile_grass) luck += 15;
+
+		return luck;
+	}
+
+	u8 mossLuck()
+	{
+		u8 luck = 0;
+		CMap@ map = getMap();
+
+		Tile tile_left = map.getTile(coords - Vec2f(map.tilesize, 0));
+		Tile tile_right = map.getTile(coords - Vec2f(-map.tilesize, 0));
+		Tile tile_above = map.getTile(coords - Vec2f(0, map.tilesize));
+		Tile tile_below = map.getTile(coords - Vec2f(0, -map.tilesize));
+
+		if (map.isTileGround(tile_below.type) || map.isTileBedrock(tile_below.type) || tile_below.type == CMap::tile_castle_moss || tile_below.type == CMap::tile_castle_back_moss) {luck += 2;}
+		if (map.isTileGround(tile_above.type) || map.isTileBedrock(tile_above.type) || tile_above.type == CMap::tile_castle_moss || tile_above.type == CMap::tile_castle_back_moss) {luck += 2;}
+		if (map.isTileGround(tile_left.type) || map.isTileBedrock(tile_left.type) || tile_left.type == CMap::tile_castle_moss || tile_left.type == CMap::tile_castle_back_moss) {luck += 2;}
+		if (map.isTileGround(tile_right.type) || map.isTileBedrock(tile_right.type) || tile_right.type == CMap::tile_castle_moss || tile_right.type == CMap::tile_castle_back_moss) {luck += 2;}
+
+		return luck;
+	}
+};
+
+void onInit(CRules@ this)
+{
+	CMap@ map = getMap();
+	if (map!is null)
+	{
+		tiles.insertLast(TileInfo(Vec2f(-1,-1), 0, map.getTile(Vec2f(0,0))));
+		for (u32 x = 0; x < map.tilemapwidth; x++)
+		{
+			for (u32 y = 0; y < map.tilemapheight; y++)
+			{
+				Vec2f coords(x*map.tilesize, y*map.tilesize);
+
+				Tile tile = map.getTile(coords);
+				Tile tile_left = map.getTile(coords-Vec2f(map.tilesize, 0));
+				Tile tile_right = map.getTile(coords-Vec2f(-map.tilesize, 0));
+				Tile tile_above = map.getTile(coords-Vec2f(0, map.tilesize));
+				Tile tile_below = map.getTile(coords-Vec2f(0, -map.tilesize));
+
+				if (map.isTileGround(tile.type) && (tile_above.type == CMap::tile_empty || (map.isTileBackground(tile_above) && tile_above.type != CMap::tile_ground_back))) 
+					tiles.insertLast(TileInfo(coords, 0, tile)); //add if there's castle/wood wall above dirt so grass grows after its destroyed
+
+				if (tile.type == CMap::tile_castle || tile.type == CMap::tile_castle_back) 
+					tiles.insertLast(TileInfo(coords, 0, tile));
+			}
+		}
+		if (!map.hasScript("grow.as")) map.AddScript("grow.as");
+	}
+
+}
+
+void onRestart(CRules@ this)
+{
+	CMap@ map = getMap();
+	tiles.clear();
+
+	if (map !is null)
+	{
+		if (!map.hasScript("grow.as")) map.AddScript("grow.as");
+	}

```

Comment:
You don't need this here because you have the same code on line 93.
```suggestion
```

Reply:
I think this one is needed since its adding a new map script, which will be removed upon loading a new map iirc

Reply:
I checked when each one is called. The one in onInit is called every time this one is called plus when the server starts for the first time.